app/controllers/concerns/hal_actions/cache.rb

```diff
@@ -0,0 +1,34 @@
+module HalActions::Cache
+  private
+
+  def index_cache_path
+    PagedCollection.new(
+      filtered(paged(sorted(scoped(resources_base)))),
+      request,
+      item_class: self.class.resource_class,
+      item_decorator: self.class.resource_representer
+    ).cache_key
+  end
+
+  def show_cache_path
+    show_resource.updated_at.utc.to_i
```

Comment:
It is not clear to me what we gain by introducing more complexity here. As it works now, `show_resource` does not call the equivalent of `#included`, so we could remove it from `show_resource`. My inclination is to benchmark it, because I honestly couldn't tell you whether it slows things down to have the `include` or to have to make 2 separate resource calls on a cache miss.